### ISSUE TITLE
Ride entrance placement seems to remember last hovered station

### ISSUE THREAD

**Issue by Broxzier** [OpenRCT2/OpenRCT2#15377]
### Operating System

Windows 10 x64

### OpenRCT2 build

OpenRCT2, v0.3.4.1 (a399108 on develop) provided by GitHub

### Describe the issue

When placing ride entrances and exits, it seems like the game remembers the last station you hovered over, and only tries to place the entrance for that one. Until you hover over another station, it simply won't place the ghost.

### Steps to reproduce

1. Build a tracked ride with two entrances
2. Try placing an entrance at one, then move the mouse to the other, like in the video
3. Notice how the game won't let you place one until you hover over the other entrance

### Attachments

https://user-images.githubusercontent.com/9705046/132563839-ab02ea03-4610-4860-84ca-40474d377fde.mp4

**Comment by TheClaudeQC** [comment]
yeah, i think that happened a lot whit me also, but when it happened first, i didn't have a Git Hub account yet, so i said, i will wait for now. So I know that one.

**Comment by spacek531** [comment]
I am pretty sure this is an original bug. I have a feeling this may have stemmed from Chris Sawyer optimizing ride entrance direction finding code for tracked ride stations (except for tower rides)

**Comment by seanmajorpayne** [comment]
I am reviewing this at the moment.

The error occurs in RideConstruction.cpp in the ride_construction_toolupdate_entrance_exit() function at this check

```
    if (gRideEntranceExitPlaceDirection == INVALID_DIRECTION)
    {
        ride_construction_invalidate_current_track();
        return;
    }
```

I'm trying to see if there is a simple addition to this check that can be made to skip it if the station is different.

**Comment by seanmajorpayne** [comment]
To be more clear, that's not where the error is caused, but that is the check passing that prevents the ghost code from running.

**Comment by spacek531** [comment]
it's caused here https://github.com/OpenRCT2/OpenRCT2/blob/e2841477bda7c504b559bf60cac8a2a7350838f3/src/openrct2/ride/RideConstruction.cpp#L1377

**Comment by seanmajorpayne** [comment]
Confirmed that this is in the original game. Would you still like the behavior modified?

**Comment by Broxzier** [comment]
Expected behaviour is that the entrance ghost pieces show up for any valid location you hover the mouse, without first having to hover over the nearby station.

**Comment by spacek531** [comment]
It's possible to remove all the code relating specifically to regular rides and only go with flat ride station code.

**Comment by seanmajorpayne** [comment]
@spacek531 Removing that mostly worked. I just needed to make sure station indexes were updated based on the current tile position, rather than set once and ignored after.

@Broxzier This should be working as you desired now.
